### PR TITLE
feat: integrate new ParserJS version

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Integrate new ParserJS version:
- use new version of `parser-js`
- update library tests and examples tests

**Related issue(s)**
Part of https://github.com/asyncapi/parser-js/issues/481

### CODE REVIEW

File: examples/integrate-with-react/package.json [hunk 1]
```diff
@@ -19,5 +19,11 @@
     "test:watch": "react-scripts test --watchAll=false",
     "eject": "react-scripts eject",
     "test:debug": "react-scripts --inspect-brk test --runInBand --no-cache"
+  },
+  "jest": {
+    "moduleNameMapper": {
+      "^nimma/legacy$": "<rootDir>/../../node_modules/nimma/dist/legacy/cjs/index.js",
+      "^nimma/(.*)": "<rootDir>/../../node_modules/nimma/dist/cjs/$1"
+    }
```
Comment (jonaslagoni) [inline examples/integrate-with-react/package.json:27]: Whats these for? 🤔

**Review by jonaslagoni — COMMENTED** [review]
Few questions ✌️

File: test/generators/typescript/__snapshots__/TypeScriptGenerator.spec.ts.snap [hunk 2]
```diff
@@ -6,12 +6,12 @@ Array [
...
   private _petType: PetType;
   private _reservedName: string;
-  private _huntingSkill: AnonymousSchema_5;
+  private _huntingSkill: AnonymousSchema_6;
 
   constructor(input: {
     petType: PetType,
     reservedName: string,
-    huntingSkill: AnonymousSchema_5,
+    huntingSkill: AnonymousSchema_6,
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply test/generators/typescript/__snapshots__/TypeScriptGenerator.spec.ts.snap:14]: I see where, we don't treat `$id` as unique id for schema. I will fix that in `parser-js` side.

File: examples/integrate-with-react/package.json (hunk 1 above)
Comment (magicmatatjahu) [reply examples/integrate-with-react/package.json:27]: https://github.com/stoplightio/spectral/issues/2002#issuecomment-994770572 :)

Comment (jonaslagoni) [reply examples/integrate-with-react/package.json:27]: Do everyone who integrates the parser, now have to do this if they use Jest? 🤔

Comment (magicmatatjahu) [reply examples/integrate-with-react/package.json:27]: Yep, I described that in this PR https://github.com/asyncapi/parser-js/pull/645 in the Readme.

File: test/generators/typescript/__snapshots__/TypeScriptGenerator.spec.ts.snap (hunk 2 above)
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply test/generators/typescript/__snapshots__/TypeScriptGenerator.spec.ts.snap:14]: Ok, It's fixed in the https://github.com/asyncapi/parser-js/pull/645 PR. I will bump parser, when we will release new version, so `huntingSkill` will to the `AnonymousSchema_5`.

**Comment by magicmatatjahu** [comment]
@jonaslagoni Done :)

**Review by jonaslagoni — APPROVED** [review]
LGTM 👍

**Pull request merged**